xgboost_ray/main.py

```diff
@@ -512,18 +517,51 @@ def _shutdown(actors: List[ActorHandle],
         queue.shutdown()
     if event:
         event.shutdown()
+    if placement_group:
+        remove_placement_group(placement_group)
+
+def _create_communication_processes():
+    # Create Queue and Event actors and make sure to colocate with driver node.
+    node_ip = ray.services.get_node_ip_address()
+    # Have to explicitly set num_cpus to 0.
+    placement_option = {"num_cpus": 0, "resources": {f"node:{node_ip}": 0.01}}
+    queue = Queue(actor_options=placement_option)  # Queue actor
+    stop_event = Event(actor_options=placement_option)  # Stop event actor
+    return queue, stop_event
+
+def _create_placement_group(cpus_per_actor, gpus_per_actor,
+                            resources_per_actor, num_actors):
+    resources_per_bundle = {"CPU": cpus_per_actor, "GPU": gpus_per_actor}
+    extra_resources_per_bundle = {} if resources_per_actor is None else \
+        resources_per_actor
+    # Create placement group for training worker colocation.
+    bundles = [{**resources_per_bundle, **extra_resources_per_bundle} for _ in range(num_actors)]
+    pg = placement_group(bundles, strategy="PACK")
+    # Wait for placement group to get created.
+    logger.debug("Waiting for placement group to start.")
+    ready = ray.wait([pg.ready()], timeout=100)
+    if ready:
+        logger.debug("Placement group has started.")
+    else:
+        raise TimeoutError("Placement group creation timed out. Make sure "
```

Comment:
Maybe you can also print available resources and required resources?

Reply:
Good point- added!